src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -0,0 +1,167 @@
+package firrtl.passes
+
+import scala.collection.mutable.{ArrayBuffer,HashMap}
+import AnalysisUtils._
+import firrtl.WrappedExpression._
+import firrtl.ir._
+import firrtl._
+import firrtl.Utils._
+import firrtl.Mappers._
+
+case class AppendableInfo(fields: Map[String,Any]) extends Info {
+  def append(a: Map[String,Any]) = this.copy(fields = fields ++ a)
+  def append(a: (String,Any)): AppendableInfo = append(Map(a))
+  def get(f: String) = fields.get(f)
+  override def equals(b: Any) = b match {
+    case i: AppendableInfo => fields - "info" == i.fields - "info"
+    case _ => false
+  }
+}
+
+object AnalysisUtils {
+
+  def getConnects(m: Module) = {
+    val connects = HashMap[String, Expression]()
+    def getConnects(s: Statement): Unit = s match {
+      case s: Connect =>
+        connects(s.loc.serialize) = s.expr
+      case s: DefNode =>
+        connects(s.name) = s.value
+      case s: Block =>
+        s.stmts foreach getConnects
+      case _ =>
+    }
```

Comment:
Also, can you explain "unapply matching"? I'm having a hard time understanding how s map getConnects actually works... It just recursively calls getConnects on each element of s? But that only neds to happen for Blocks? So I personally feel it's more clear to do the Block stuff rather than s map getConnects, unless I'm missing something...